### PR TITLE
Improve linux activate

### PR DESCRIPTION
**Pull request by MestreLion**
As discussed in #36

### CODE REVIEW

**Comment by Kalmat** [comment]
Hi! The PR failed because of the typing checks. As I mentioned before, this all about of typing is like black magic to me. In fact, I had to create a "dummy" repo (PyWinCtl-Typed) to check there before commiting to master branch and get it full of useless commits (unfortunately it's a try-fail process for me)

In this case, I think the workaround (since we are accessing a private method) can be something like this:

`EWMH._setProperty('_NET_ACTIVE_WINDOW', [source, Xlib.X.CurrentTime, self._hWnd.id], self._hWnd)  # pyright: ignore[reportUnknownMemberType, reportGeneralTypeIssues]  # We expect an error here`

Besides... what is this???  `# noqa : W0212`. Never seen it before!

File: src/pywinctl/_pywinctl_linux.py [hunk 1]
```diff
@@ -609,19 +609,24 @@ def hide(self, wait: bool = False) -> bool:
...
         """
         if "arm" in platform.platform():
             EWMH.setWmState(self._hWnd, ACTION_UNSET, STATE_BELOW, STATE_NULL)
             EWMH.display.flush()
             EWMH.setWmState(self._hWnd, ACTION_SET, STATE_ABOVE, STATE_FOCUSED)
         else:
-            EWMH.setActiveWindow(self._hWnd)
+            # https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#sourceindication
+            source = 2 if user else 1
+            EWMH._setProperty('_NET_ACTIVE_WINDOW',  # noqa : W0212
```
Comment (MestreLion) [inline src/pywinctl/_pywinctl_linux.py:627]: See? I told you that accessing a private method would give us some headaches :-P

Not sure why pyright and mypy didn't honor the `# noqa : W0212` remark.
Or maybe we just need to add `EWMH._setProperty` to your ewmh typing file?

**Review by MestreLion — COMMENTED** [review]
Need to update the ewmh typing file

**Comment by MestreLion** [comment]
> In this case, I think the workaround (since we are accessing a private method) can be something like this:
> 
> `EWMH._setProperty('_NET_ACTIVE_WINDOW', [source, Xlib.X.CurrentTime, self._hWnd.id], self._hWnd) # pyright: ignore[reportUnknownMemberType, reportGeneralTypeIssues] # We expect an error here`

Those warnings are not about accessing a private method, but about accessing an _unknown_ method with an _unknown_ return type.

No need to mute _that_ warning, at least not _yet_: it might suffice to add `def _setProperty(self, _type, data, win=None, mask=None) -> None:` to [typings/ewmh/ewmh.pyi](../blob/master/typings/ewmh/ewmh.pyi)

> Besides... what is this??? `# noqa : W0212`. Never seen it before!

**_That's_** the marker for flake8/pylint/etc to avoid a warning for accessing a private method ;-)

https://pylint.pycqa.org/en/latest/user_guide/messages/warning/protected-access.html

File: src/pywinctl/_pywinctl_linux.py (hunk 1 above)
Comment (Kalmat) [reply src/pywinctl/_pywinctl_linux.py:628]: HAHAHAHA! Yes, you told me... but there was no other option!

I'm having my own hell trying to commit other changes. I have done more than 30 commits this evening, without success... still trying. As I told you, all this about typing is like black magic to me. I'm not saying it is not good, or necessary, or useful or even highly recommendable... but trust me that right now even the last of my cells deeply hate it.

**Comment by MestreLion** [comment]
It seems adding the method to ewmh.pyi was the solution to _that_ problem.

But... wow, they require _full_ typing, as in `list` is not acceptable? Geez!!! Ok, will fix that

File: src/pywinctl/_pywinctl_linux.py (hunk 1 above)
Comment (MestreLion) [reply src/pywinctl/_pywinctl_linux.py:628]: Well, using typing is one thing, _requiring_ full 100% coverage for _all_ methods, including external libraries, _that's_ insane! lol

**Review by Kalmat — APPROVED** [review]

**Comment by MestreLion** [comment]
Finally passed the tests!!!

Those last errors were because I added a missing `) -> bool:` to `LinuxWindow.activate()`, which opened a pandora box: the return type was missing from `isActive()`, which in turn also had a returned value that wasn't typed as bool (what else woyld `==` be???). Geez!!!

**Pull request merged**